### PR TITLE
Allow sorting by institution

### LINKED ISSUE #1355: Campaign programs list should be sortable by Institution

### PR DESCRIPTION
**Pull request by alehed**
Currently the data-sort key is not correct yet…

Closes #1355

### CODE REVIEW

**Comment by alehed** [comment]
I'm currently at a road-block with this, since locally I managed to break the sorting functionality for the table in question (probably via something that is not tracked with source control).

Since I am new to javascript maybe it is best if somebody else who understands what is going on can implement this. Feel free to use my branch as the starting point, the required changes seem to be trivial.

Here is my local behavior of the current master (1c01dfb3c), note how manipulating the last table does not work:
wikiedusorting

**Comment by ragesoss** [comment]
@alehed the diff from the last time we added new sorting functionality may be helpful: https://github.com/WikiEducationFoundation/WikiEduDashboard/commit/01636d62d79a8b89ba8360ca485a9662d5fba523#diff-2812d5eeee31fe25d48dea8be520534a

There are several different aspects required to make it work: the javascript needs to be looking for the right column key as a sortable one, the column header needs to have the right css class to mark it as sortable, and the row must have the relevant data key to indicate what to sort by.

**Comment by alehed** [comment]
I see… I was missing the td class.

Added a test and a fix for the regression in 'a465db6'. Is is considered good style to have small helper functions in the test file itself?

Feel free to squash-merge this.

File: spec/features/campaign_programs_spec.rb [hunk 1]
```diff
@@ -6,10 +6,15 @@
   let(:user)  { create(:user) }
   let(:campaign) { create(:campaign) }
   let(:course) { create(:course) }
```
[the remark this answers is not in the document]
Comment (alehed) [reply spec/features/campaign_programs_spec.rb]: Ok, I'll fix that

**Comment by ragesoss** [comment]
> Is it considered good style to have small helper functions in the test file itself?

Yep!

On a related note, I try to avoid xpath noise as much as possible in capybara tests, because it can be pretty tedious to reason about and to change, and being so precise often makes it pretty brittle. It's a not a big deal, and sometimes there is no cleaner way to write the test — I don't know whether or not that's the case here — but in general, the fewer references to selectors, the better.

**Comment by alehed** [comment]
Any ideas on how I can avoid it here? I basically just need to test for the first cell of the given row.
I could match regular expressions agains the entire html page, but I consider that less beautiful (and precise).

**Comment by alehed** [comment]
Should I change `//table/tbody/tr[1]/td[1]` to `//tr[1]/td[1]` or even `//tr[1]`?

**Comment by ragesoss** [comment]
Yeah, there might not be a good way in this case; nothing comes to mind.

In the React portions of the frontend, we could test it at the unit level instead. But since we're doing jquery DOM manipulation, maybe there's no better option.

**Comment by alehed** [comment]
Ok, I'll change it to `//tbody/tr[1]`.

**Pull request merged**